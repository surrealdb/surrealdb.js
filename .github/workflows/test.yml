name: test

on:
  push:
  pull_request:

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Setup node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
      - name: Setup deno
        uses: denoland/setup-deno@v1
        with:
          deno-version: v1.x
      - name: Setup bun
        uses: antongolub/action-setup-bun@v1
        with:
          bun-version: 0.1.13

      - uses: actions/checkout@v3

      - name: Build NPM package
        run: deno task build

      - name: Start SurrealDB
        run: sh ./test/e2e/scripts/docker-restart.sh

      - name: Run Deno
        run: deno test -A --trace-ops ./test/e2e/deno.js
      - name: Run Node
        run: node ./test/e2e/node.js
      - name: Run Bun
        run: bun run ./test/e2e/bun.js

      - name: Stop SurrealDB
        run:  sh ./test/e2e/scripts/docker-stop.sh
