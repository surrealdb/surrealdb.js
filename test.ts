import { RecordId } from "./src/library/data/recordid.ts";
import { Duration } from "./src/library/data/duration.ts";
import { UUID } from "./src/library/data/uuid.ts";
import { Decimal } from "./src/library/data/decimal.ts";
import { Surreal } from "./src/index.ts";
import { encode } from "./src/library/data/cbor.ts";
import { GeometryCollection, GeometryLine, GeometryPoint } from "./src/library/data/geometry.ts";
// import { recordId } from "./src/library/orm/types.ts";

const surreal = new Surreal();
await surreal.connect('ws://127.0.0.1:8000/rpc');
await surreal.use({
	namespace: 'test',
	database: 'test',
});

// console.log(
// 	await surreal.query<[unknown[]]>(
// 		"[$d, $r, $c, $u, $dur, $none, NONE, $a, <string> $a, $nan, $geo]",
// 		{
// 			d: new Date(),
// 			r: new RecordId('person', 'tobie'),
// 			c: new RecordId('recording', [
// 				new Date(),
// 				'London',
// 				new Uuid(),
// 				{
// 					temparature: new Decimal("10.00003")
// 				}
// 			]),
// 			u: new Uuid(),
// 			dur: new Duration('1d20m'),
// 			none: undefined,
// 			a: 9223372036854775807n,
// 			nan: NaN,
// 			geo: new GeometryCollection([
// 				new GeometryPoint([1, 2]),
// 				new GeometryLine([ new GeometryPoint([1, 2]), new GeometryPoint([3, 4]) ])
// 			])
// 		}
// 	)
// )

const uuid = new Duration("2w6d100ms");
console.log(
	456,
	await surreal.query<[UUID, string]>(
		"$uuid; <string> $uuid",
		{
			uuid
		}
	).then(([u, a]) => [u.toString(), a])
)

// function bufferToHex (buffer: ArrayBuffer) {
//     return [...new Uint8Array (buffer)]
//         .map (b => b.toString (16).padStart (2, "0"))
//         .join ("");
// }

// console.log(bufferToHex(encode(new RecordId('recording', [
// 	new Date(),
// 	'London',
// 	new Uuid(),
// 	{
// 		temparature: new Decimal("10.00003")
// 	}
// ]))))

// const rid = recordId('test');
// console.log(rid.parse(new RecordId('tesst', 123) ))

// console.log()
