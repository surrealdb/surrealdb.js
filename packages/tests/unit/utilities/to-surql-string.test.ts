import { describe, expect, test } from "bun:test";
import { toSurqlString } from "surrealdb";
import { createMockValue } from "../__helpers__";

describe("toSurqlString()", () => {
	test("match snapshot", () => {
		const string = toSurqlString(createMockValue());

		expect(string).toMatchSnapshot();
	});
});
