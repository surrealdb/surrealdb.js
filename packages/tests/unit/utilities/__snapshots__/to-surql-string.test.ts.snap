// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`toSurqlString() match snapshot 1`] = `"{ "rid": r"⟨some:thing⟩:under_score", "id_looks_like_number": r"⟨some:thing⟩:⟨123⟩", "id_almost_a_number": r"⟨some:thing⟩:1e23", "id_is_a_number": r"⟨some:thing⟩:123", "str_rid": r"⟨some:thing⟩:under_score", "rng_rid": bla:a..z, "range": r"bla:a"..z, "range_unbounded": ..z, "dec": 3.333333, "dur": 1d2h, "geo": { "type": s"GeometryCollection", "geometries": [ { "type": s"Point", "coordinates": [ 1, 2 ] }, { "type": s"MultiPolygon", "coordinates": [ [ [ [ 1, 2 ], [ 3, 4 ], [ 1, 2 ] ], [ [ 5, 6 ], [ 7, 8 ], [ 5, 6 ] ] ] ] }, { "type": s"Polygon", "coordinates": [ [ [ 1, 2 ], [ 3, 4 ], [ 1, 2 ] ], [ [ 5, 6 ], [ 7, 8 ], [ 5, 6 ] ] ] } ] }, "tb": ⟨some super _ cool table⟩, "uuid": u"92b84bde-39c8-4b4b-92f7-626096d6c4d9", "date": d"2024-05-06T17:44:57.085Z", "file": f"my\\\\ bucket:/file/name", "undef": NONE, "null": NULL, "num": 123, "float": 123.456, "true": true, "false": false, "string": s"I am a string" }"`;
